Sparkle/SUBinaryDeltaApply.m

```diff
@@ -15,38 +15,84 @@
 #include <stdlib.h>
 #include <xar/xar.h>
 
-static void applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
+static BOOL applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
 {
     NSString *patchFile = temporaryFilename(@"apply-binary-delta");
     xar_extract_tofile(x, file, [patchFile fileSystemRepresentation]);
     const char *argv[] = {"/usr/bin/bspatch", [sourceFilePath fileSystemRepresentation], [destinationFilePath fileSystemRepresentation], [patchFile fileSystemRepresentation]};
-    bspatch(4, (char **)argv);
+    BOOL success = (bspatch(4, (char **)argv) == 0);
     unlink([patchFile fileSystemRepresentation]);
+    return success;
 }
 
 int applyBinaryDelta(NSString *source, NSString *destination, NSString *patchFile)
 {
-    xar_t x = xar_open([patchFile UTF8String], READ);
+    xar_t x = xar_open([patchFile fileSystemRepresentation], READ);
     if (!x) {
-        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile UTF8String]);
+        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile fileSystemRepresentation]);
         return 1;
     }
+    
+    SUBinaryDeltaMajorVersion majorDiffVersion = FIRST_DELTA_DIFF_MAJOR_VERSION;
+    SUBinaryDeltaMinorVersion minorDiffVersion = FIRST_DELTA_DIFF_MINOR_VERSION;
```

Comment:
I think minor version is not needed, since it doesn't affect behavior. New, optional features can be added as new keys.